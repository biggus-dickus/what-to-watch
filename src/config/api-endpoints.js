export const BASE_HOST = `https://es31-server.appspot.com`;

export const ApiEndpoint = {
  BASE_URL: `${BASE_HOST}/wtw`,
  COMMENTS: `/comments`,
  FILMS: `/films`,
  FAVOURITE: `favorite`,
  LOGIN: `/login`
};
