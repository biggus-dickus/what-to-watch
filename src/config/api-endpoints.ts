export const BASE_HOST: string = `https://es31-server.appspot.com`;

export enum ApiEndpoint {
  BASE_URL = 'https://es31-server.appspot.com/wtw',
  COMMENTS = '/comments',
  FILMS = '/films',
  FAVOURITE = 'favorite',
  LOGIN = '/login',
  PROMO = '/films/promo'
}
