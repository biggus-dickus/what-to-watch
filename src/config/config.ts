// How much time to wait in ms before playing a film preview
export const HOVER_DELAY = 1000;

// How many films to render in "More like this" section (film page)
export const MORE_LIKE_THIS_LIMIT = 4;

// Navigation tabs on film page
export const NAV_OVERVIEW_ID = `overview`;
export const NAV_DETAILS_ID = `details`;
export const NAV_REVIEWS_ID = `reviews`;
export const filmTabs = [
  {
    id: NAV_OVERVIEW_ID,
    text: `Overview`
  },
  {
    id: NAV_DETAILS_ID,
    text: `Details`
  },
  {
    id: NAV_REVIEWS_ID,
    text: `Reviews`
  },
];
