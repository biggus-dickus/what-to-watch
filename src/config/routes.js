const Route = Object.freeze({
  ADD_REVIEW: `/film/:id/review`,
  FILM: `/film/:id`,
  INDEX: `/`,
  MY_LIST: `/mylist`,
  SIGN_IN: `/login`
});

export default Route;

export const isPrivateRoute = (pathname) => {
  const myListRe = /\/mylist/;
  const addReviewRe = /\/film\/[\d]\/review/;

  return myListRe.test(pathname) || addReviewRe.test(pathname);
};
