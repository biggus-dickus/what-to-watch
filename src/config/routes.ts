enum Route {
  ADD_REVIEW = '/film/:id/review',
  FILM = '/film/:id',
  INDEX = '/',
  MY_LIST = '/mylist',
  SIGN_IN = '/login'
}

export default Route;

export const isPrivateRoute = (pathname: string): boolean => {
  const myListRe = /\/mylist/;
  const addReviewRe = /\/film\/[1-9][\d]*\/review/;

  return myListRe.test(pathname) || addReviewRe.test(pathname);
};
