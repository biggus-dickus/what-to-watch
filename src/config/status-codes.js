export const StatusCode = Object.freeze({
  OK: 200,
  NOT_MODIFIED: 304,
  BAD_REQUEST: 400,
  FORBIDDEN: 403,
  NOT_FOUND: 404,
  INTERNAL_SERVER_ERROR: 500,
  SERVICE_UNAVAILABLE: 503
});
