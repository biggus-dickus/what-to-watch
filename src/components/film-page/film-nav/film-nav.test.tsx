import * as React from 'react';
import {StaticRouter as Router} from 'react-router';
import * as renderer from 'react-test-renderer';

import {filmTabs} from '../../../config/config';
import FilmNav from './film-nav';


const props = {
  activeTabId: filmTabs[0].id,
  tabs: filmTabs,
  onTabClick: jest.fn()
};

it(`<FilmNav /> should render correctly`, () => {
  expect(renderer.create(<Router><FilmNav {...props} /></Router>).toJSON()).toMatchSnapshot();
});
