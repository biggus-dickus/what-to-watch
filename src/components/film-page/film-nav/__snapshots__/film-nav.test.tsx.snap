// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmNav /> should render correctly 1`] = `
<nav
  className="movie-nav movie-card__nav"
>
  <ul
    className="movie-nav__list"
  >
    <li
      className="movie-nav__item movie-nav__item--active"
    >
      <a
        className="movie-nav__link"
        href="/#overview"
        onClick={[Function]}
      >
        Overview
      </a>
    </li>
    <li
      className="movie-nav__item"
    >
      <a
        className="movie-nav__link"
        href="/#details"
        onClick={[Function]}
      >
        Details
      </a>
    </li>
    <li
      className="movie-nav__item"
    >
      <a
        className="movie-nav__link"
        href="/#reviews"
        onClick={[Function]}
      >
        Reviews
      </a>
    </li>
  </ul>
</nav>
`;
